doc: add section about new lines in mkDerivation (#156713)
NixOS/nixpkgs: doc/contributing/coding-conventions.chapter.md

  })
  ```

> - Between first level not string-like variables in `mkDerivation` like `buildInputs`, `patches` or `meta` should be an empty line to make the code less cramped and more clear:

Reviewer: What are "not string-like" variables? What are "string-like" variables? Maybe call it "compound attributes" (because in a pure language we don't even have variables)?

I disagree on any kind of rationale for formatting decisions, because this will by nature of being an aesthetic judgment always be subject to bikeshedding. You motivate establishing code styling rules with preempting these fruitless discussions. Let's do just that and not offer attack surface.

Maybe:
```suggestion
- Between top-level compound attributes to `mkDerivation`, such as `buildInputs`, `patches` or `meta`, there should be an empty line:
```
